### PR TITLE
test(dm): add test for json

### PR DESCRIPTION
**Pull request by xiongjiwei**
### What problem does this PR solve?

Issue Number: ref #4159

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: dm/tests/all_mode/data/db3.prepare.sql [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+insert into t_extract(j)values ('1');
+insert into t_extract(j)values ('3.14');
+insert into t_extract(j)values ('{"a":1, "b": {"a":1}, "c": "string"}');
+
+create table t_funcs
+(
+    id int primary key auto_increment,
+    j  json,
+    j1 json as (json_extract(j, '$.a')),
+    j2 json as (json_extract(j1, '$.b')),
```
Comment (lance6716) [inline dm/tests/all_mode/data/db3.prepare.sql:29]: Does it cover all valid JSON function which can be used as generating expression?

Comment (xiongjiwei) [reply dm/tests/all_mode/data/db3.prepare.sql:44]: it covered now

File: dm/tests/all_mode/run.sh [hunk 2]
```diff
@@ -375,6 +375,38 @@ function test_regexpr_router() {
 	echo "[$(date)] <<<<<< finish test_regexpr_router $1 >>>>>>"
 }
 
+function test_json_expression() {
```
Comment (lance6716) [inline dm/tests/all_mode/run.sh:378]: this function should be called at around line 414.

And I'll help to check CI if it fails

File: dm/tests/all_mode/data/db3.prepare.sql [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    j5 json as (json_merge_patch(b, '{"k": "v"}')),
+    j6 json as (json_merge(b, '{"k": "v"}')),
+    j7 json as (json_set(b, '$.k', 'v')),
+    j8 json as (json_insert(b, '$.k', 'v')),
+    j9 json as (json_replace(b, '$.k', 'v')),
+    j10 json as (json_remove(b, '$.k')),
+    j11 json as (json_contains(b, '{"k": "v"}')),
+    j12 json as (json_contains_path(b, 'one', '$.k')),
+    j13 json as (json_array_append(b, '$.k', 'v')),
+
```
Comment (lance6716) [inline dm/tests/all_mode/data/db3.prepare.sql:24]:
```suggestion
```

File: dm/tests/all_mode/data/db3.prepare.sql [hunk 4]
```diff
@@ -0,0 +1,57 @@
+drop database if exists `all_mode`;
+create database `all_mode`;
+use `all_mode`;
+
+create table t_extract
```
Comment (lance6716) [inline dm/tests/all_mode/data/db3.prepare.sql:5]: please do ALTER TABLE ADD COLUMN with all these JSON function of the two tables in db3.increment.sql, and insert some data

**Review by lance6716 — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**